In [1]:
import pandas as pd # for data manupulation or analysis
import numpy as np # for numeric calculation
import matplotlib.pyplot as plt # for data visualization
import seaborn as sns # for data visualization

In [2]:
from sklearn.datasets import load_breast_cancer
cancer_dataset = load_breast_cancer()

In [3]:
# create datafrmae
cancer_df = pd.DataFrame(np.c_[cancer_dataset['data'],cancer_dataset['target']],columns = np.append(cancer_dataset['feature_names'], ['target']))

In [4]:
cancer_df.tail(10)

mean radius  mean texture  ...  worst fractal dimension  target
559        11.51         23.93  ...                  0.08732     1.0
560        14.05         27.15  ...                  0.08321     1.0
561        11.20         29.37  ...                  0.05905     1.0
562        15.22         30.62  ...                  0.14090     0.0
563        20.92         25.09  ...                  0.09873     0.0
564        21.56         22.39  ...                  0.07115     0.0
565        20.13         28.25  ...                  0.06637     0.0
566        16.60         28.08  ...                  0.07820     0.0
567        20.60         29.33  ...                  0.12400     0.0
568         7.76         24.54  ...                  0.07039     1.0

[10 rows x 31 columns]

In [5]:
cancer_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   mean radius              569 non-null    float64
 1   mean texture             569 non-null    float64
 2   mean perimeter           569 non-null    float64
 3   mean area                569 non-null    float64
 4   mean smoothness          569 non-null    float64
 5   mean compactness         569 non-null    float64
 6   mean concavity           569 non-null    float64
 7   mean concave points      569 non-null    float64
 8   mean symmetry            569 non-null    float64
 9   mean fractal dimension   569 non-null    float64
 10  radius error             569 non-null    float64
 11  texture error            569 non-null    float64
 12  perimeter error          569 non-null    float64
 13  area error               569 non-null    float64
 14  smoothness error         5

In [6]:
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score

In [7]:
cancer_df_meanone=cancer_df[["mean radius","mean texture","mean perimeter","mean area","mean concave points","target"]]
# cancer_df_meanone=cancer_df[["mean radius","mean texture","mean perimeter","mean area","mean smoothness", "mean compactness","mean concave points", "target"]]
# cancer_df_meanone=cancer_df[["mean radius","mean texture","mean perimeter","mean area","mean compactness","target"]]
# cancer_df_meanone=cancer_df[["mean radius","mean texture","mean perimeter","mean area","target"]] - 94%
# cancer_df_meanone=cancer_df[["mean radius","mean texture","mean perimeter","mean area","mean concave points","target"]] - 95%

print(cancer_df_meanone.tail(7))

     mean radius  mean texture  ...  mean concave points  target
562        15.22         30.62  ...              0.09429     0.0
563        20.92         25.09  ...              0.14740     0.0
564        21.56         22.39  ...              0.13890     0.0
565        20.13         28.25  ...              0.09791     0.0
566        16.60         28.08  ...              0.05302     0.0
567        20.60         29.33  ...              0.15200     0.0
568         7.76         24.54  ...              0.00000     1.0

[7 rows x 6 columns]


In [8]:
# input variable
X_mean = cancer_df_meanone.drop(['target'], axis = 1)
print(X_mean.tail(6))

# output variable
y_mean = cancer_df_meanone['target']
print(y_mean.tail(5))

from sklearn.model_selection import train_test_split
X_train_mean, X_test_mean, y_train_mean, y_test_mean = train_test_split(X_mean, y_mean, test_size = 0.2, random_state= 5)


from sklearn.metrics import confusion_matrix, classification_report, accuracy_score

# Logistic Regression
from sklearn.linear_model import LogisticRegression
lr_classifier = LogisticRegression(random_state = 51, penalty = 'none',max_iter=1000)
lr_classifier.fit(X_train_mean, y_train_mean)

     mean radius  mean texture  mean perimeter  mean area  mean concave points
563        20.92         25.09          143.00     1347.0              0.14740
564        21.56         22.39          142.00     1479.0              0.13890
565        20.13         28.25          131.20     1261.0              0.09791
566        16.60         28.08          108.30      858.1              0.05302
567        20.60         29.33          140.10     1265.0              0.15200
568         7.76         24.54           47.92      181.0              0.00000
564    0.0
565    0.0
566    0.0
567    0.0
568    1.0
Name: target, dtype: float64


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='none',
                   random_state=51, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [9]:
y_pred_lr_mean = lr_classifier.predict(X_test_mean)
print("Accuracy by logistic regression for 4 parameters : ",accuracy_score(y_test_mean, y_pred_lr_mean))

Accuracy by logistic regression for 4 parameters :  0.956140350877193


##getting accuracy of 95.6% with logistic regression with 5 attributes.

In [10]:
z=lr_classifier.predict([[7,24,50,182,0]])
z

array([1.])

#saving the model

In [16]:
## Pickle
import pickle
 
# save model
pickle.dump(lr_classifier, open('breast_cancer_detector.pickle', 'wb'))

# load model
breast_cancer_detector_model = pickle.load(open('breast_cancer_detector.pickle', 'rb'))
 
# predict the output
y_pred = breast_cancer_detector_model.predict(X_test_mean)
# print(y_pred)

# show the accuracy
print('Accuracy of linear regression model = ',accuracy_score(y_test_mean, y_pred_lr_mean))

Accuracy of linear regression model =  0.956140350877193


In [ ]:
# from google.colab import drive
# drive.mount('/content/drive')